packages/core/test/integration/fs/project-file.it.spec.ts

```diff
@@ -0,0 +1,131 @@
+import { Stats } from 'fs';
+
+import { testInjector } from '@stryker-mutator/test-helpers';
+import { expect } from 'chai';
+
+import { FileSystem, ProjectFile } from '../../../src/fs/index.js';
+import { resolveFromRoot } from '../../helpers/test-utils.js';
+
+import { CleanupFileSystem } from './cleanup-file-system.js';
+
+const resolveTestResource = resolveFromRoot.bind(undefined, 'testResources', 'project-file');
+const TARGET_DIRECTORY = 'temp';
```

Comment:
could we use the OS temp directory here?

```ts
import fs from 'fs';

const targetDirectory = await fs.promises.mkdtemp('stryker-project-file-test');
```

https://nodejs.org/api/fs.html#fspromisesmkdtempprefix-options